modules/behaviors/DataBehavior.js

```diff
@@ -0,0 +1,305 @@
+        var singleResultModel = this.privateCollection.at(0);
+        if (_.isString(singleResultProperty)) {
+          return singleResultModel.get(singleResultProperty);
+        }
+        return singleResultModel.toJSON();
+      } else {
+        throw new Error('Multiple results found, but single result expected: ' + JSON.stringify(this.privateCollection.toJSON()));
+      }
+    },
+
+    /**
+     * Retrieves the ids for this data object and passes them off to the private collection's trackAndPull() method.
+     * @method pull
+     */
+    pull: function() {
+      var thisDataBehavior = this;
+      this.__getIds()
+        .then(function(ids) {
+          return thisDataBehavior.privateCollection.trackAndPull(ids);
+        });
+    },
+
+    /**
+     * Retrieves the ids for this data object and passes them off to the private collection's trackAndFetch() method.
+     * @method fetch
+     */
+    fetch: function() {
+      var thisDataBehavior = this;
+      this.__getIds()
+        .then(function(ids) {
+          return thisDataBehavior.privateCollection.trackAndFetch(ids);
+        });
+    },
+
+    /**
+     * Retrieves the ids for this data object and passes them off to the private collection to track and then does a
+     * pull or a fetch based on the alwaysFetch property.  (pull is default if always fetch is true then it fetches instead).
+     * @method retrieve
+     */
+    retrieve: function() {
+      if (this.__alwaysFetch) {
+        return this.fetch();
+      } else {
+        return this.pull();
+      }
+    },
+
+    /**
+     * @method __getIds
+     * @return {$.Deferred.Promise} a jquery deferred promise that resolves to the ids to track in the private collection
+     *                              or rejects with the error message.
+     * @private
+     */
+    __getIds: function() {
+      var idsDeferred = $.Deferred();
+      var ids = this.__ids;
+      var normalizedIds = normalizeIds(ids);
+      if (!_.isUndefined(normalizedIds)) {
+        idsDeferred.resolve(normalizedIds);
+      } else if (_.isFunction(this.__ids)) {
+        ids = this.__ids(this.__cache);
+        normalizedIds = normalizeIds(ids);
+        if (!_.isUndefined(normalizedIds)) {
+          idsDeferred.resolve(normalizedIds);
+        } else if (!_.isUndefined(ids) && _.isFunction(ids.promise)) {
+          idsDeferred = ids;
+        } else {
+          idsDeferred.resolve([]);
+        }
+      } else if (_.isString(this.__ids.property)) {
+        var context;
+        if (_.isUndefined(this.__ids.context)) {
+          context = this.view;
+        } else if (_.isFunction(this.__ids.context)) {
+          context = this.__ids.context();
+        } else if (_.isString(this.__ids.context)) {
+          context = _.result(this, this.__ids.context);
+        } else if (_.isObject(this.__ids.context)) {
+          context = this.__ids.context;
+        } else if (_.isFunction(this.__ids.context)) {
```

Comment:
ah yep, I just assumed you were referring to line 267 instead of 263. You are correct, it's the exact same line and in the wrong place this time.